pyproject.toml

```diff
@@ -309,6 +310,13 @@ ignore_messages = "^.*(Unknown target name|Undefined substitution referenced)[^`
 ignore_errors = true
 scripts_are_modules = true
 strict = false
+modules = ["mne"]
+# Avoid the conflict between mne/__init__.py and mne/__init__.pyi by ignoring the former
+exclude = '^mne/(beamformer|channels|commands|datasets|decoding|export|forward|gui|html_templates|inverse_sparse|io|minimum_norm|preprocessing|report|simulation|source_space|stats|time_frequency|utils|viz)?/?__init__\.py$'
+
+[[tool.mypy.overrides]]
+module = ['scipy.*']
+ignore_missing_imports = true
```

Comment:
@hoechenberger did have to add this because of this failure, though

https://github.com/mne-tools/mne-python/actions/runs/8805892959/job/24169465941?pr=12569#step:6:12

With this change things are okay:

https://github.com/mne-tools/mne-python/actions/runs/8805919692/job/24169549047?pr=12569#step:6:12

Not sure why it happens on GHA but not Azure. Azure installs a lot more stuff and runs 3.11 instead of 3.12, maybe there's something wrong with the type stubs in SciPy or something for 3.12 (?). In any case I think this should be safe enough?

Reply:
I don't have the capacity to look into this now, but if it works, then let's just keep it like this for now :)